xarray/core/computation.py

```diff
@@ -1137,3 +1137,81 @@ def where(cond, x, y):
                        join='exact',
                        dataset_join='exact',
                        dask='allowed')
+
+
+def _differentiate_variable(variable, coord, edge_order):
+    """ Compute the gradient along 1 dimension for Variable.
+    variable, coord: Variable
+    """
+    from .variable import Variable
+
+    result_array = duck_array_ops.gradient(
+        variable.data, coord.data, edge_order=edge_order,
+        axis=variable.get_axis_num(coord.dims[0]))
+    return Variable(variable.dims, result_array)
+
+
+def differentiate(dataarray, coord, edge_order=1):
+    """ Compute the gradient of the dataarray with the second order accurate
+    central differences.
+
+    Parameters
+    ----------
+    dataarray: xr.DataArray
+        Target array
+    coord: str
+        The coordinate to be used to differentiate the array.
+    edge_order: 1 or 2. Default 1
+        N-th order accurate differences at the boundaries.
+
+    Returns
+    -------
+    differentiate: DataArray or a sequence of DataArrays
+
+    See also
+    --------
+    numpy.gradient: corresponding numpy function
+
+    Examples
+    --------
+
+    >>> da = xr.DataArray(np.arange(12).reshape(4, 3), dims=['x', 'y'],
+    ...                   coords={'x': [0, 0.1, 1.1, 1.2]})
+    >>> da
+    <xarray.DataArray (x: 4, y: 3)>
+    array([[ 0,  1,  2],
+           [ 3,  4,  5],
+           [ 6,  7,  8],
+           [ 9, 10, 11]])
+    Coordinates:
+      * x        (x) float64 0.0 0.1 1.1 1.2
+    Dimensions without coordinates: y
+    >>>
+    >>> xr.differentiate(da, 'x')
+    <xarray.DataArray (x: 4, y: 3)>
+    array([[30.      , 30.      , 30.      ],
+           [27.545455, 27.545455, 27.545455],
+           [27.545455, 27.545455, 27.545455],
+           [30.      , 30.      , 30.      ]])
+    Coordinates:
+      * x        (x) float64 0.0 0.1 1.1 1.2
+    Dimensions without coordinates: y
+    """
+    from .dataarray import DataArray
+
+    if not isinstance(dataarray, DataArray):
+        raise TypeError(
+            'Only DataArray is supported. Given {}.'.format(type(dataarray)))
```

Comment:
I'm a little confused here. You wrote an implementation for Dataset.differentiate, too, and here is a duplicate version of DataArray.differentiate.

Reply:
👍 I don't think we need the separate function.